File: Modules/Common/Techs/SIE.Web.Tech/Stations/Commands/StationItemImportCommand.cs
using SIE.Common.ImportHelper;
using SIE.Tech.Stations.ImportStationItem;
using SIE.Web.Command;
using SIE.Web.Common.Import.Commands;
using System;
using System.Data;

namespace SIE.Web.Tech.Stations.Commands
{
    /// <summary>
    /// 工位物料导入
    /// </summary>
    [JsCommand("SIE.Web.Tech.Stations.Commands.StationItemImportCommand")]
    public class StationItemImportCommand : ImportCommandBase
    {
        /// <summary>
        /// 导入完成回调方法
        /// </summary>
        /// <returns>导入结果</returns>
        protected override ImportCompleted GetImportCompleted()
        {
            return (DataRow[] drSuccess, DataRow[] drFailed) =>
            {
            };
        }

        /// <summary>
        /// 获取导入处理类型
        /// </summary>
        /// <returns>工位物料导入处理类</returns>
        protected override Type GetImportHandleType()
        {
            return typeof(ImportStationItemWebHandle);
        }
    }
}

File: Modules/Common/Techs/SIE.Web.Tech/Stations/Commands/StationItemImportCommand.js
SIE.defineCommand('SIE.Web.Tech.Stations.Commands.StationItemImportCommand', {
    extend: 'SIE.Web.Common.Import.Commands.ImportCommandBase',
    meta: { text: "导入", group: "business", iconCls: "icon-Upload icon-green" },
    canExecute: function (view) {
        if (view.getParent() == null || view.getParent().getCurrent() == null) return false;
        return true;
    }
});